libdnf5/rpm/package_query.cpp

```diff
@@ -2471,7 +2505,8 @@ static void filter_first_sorted_by(
     int i;
     for (i = 0; i < samename.size(); ++i) {
         Solvable * considered = pool.id2solvable(samename[i]);
-        if (!highest || highest->name != considered->name || highest->arch != considered->arch) {
+        if (!highest || highest->name != considered->name ||
+            (grouping == FilterGrouping::BY_NAME_ARCH && (highest->arch != considered->arch))) {
```

Comment:
Only  note with no significant impact on the code. It adds one boolean test which is fast,  but for each query this test is performed for each name in sets  (Fedora has something like 40 000 package  names).

Reply:
Hmm, the only alternative I can think of is duplicating the function or at least the for-loop. It should be only one additional AND instruction though?